asmcomp/arm64/emit.mlp

```diff
@@ -1053,8 +1073,33 @@ let emit_instr env i =
 
 (* Emission of an instruction sequence *)
 
-let rec emit_all env i =
-  if i.desc = Lend then () else (emit_instr env i; emit_all env i.next)
+(* for debugging instr_size errors *)
+let emit_instr_debug env i =
+  let lbl = new_label () in
+  `{emit_label lbl}:\n`;
+  emit_instr env i;
+  let sz = Size.instr_size env.f i.desc * 4 in
+  `	.ifne (. - {emit_label lbl}) - {emit_int sz}\n`;
+  `	.error \"Emit.instr_size: instruction length mismatch\"\n`;
+  `	.endif\n`
```

Comment:
I suppose I have a different idea of fragility here! I'd really like to see this or something like it run at least in CI, so that we actually detect mistakes being introduced in this area in future, rather than waiting for user reports.

Perhaps a middle ground might be to blow up only when the compiler *under*-estimates the size of an instruction sequence, since that case may actually cause miscompilations / assembler failures?